fhir-server/src/main/java/com/ibm/fhir/server/filter/rest/FHIRRestServletFilter.java

```diff
@@ -302,16 +302,23 @@ private void computeReturnPref(FHIRRequestContext context) throws FHIRException
      */
     private String getHeaderTokenValue(FHIRRequestContext context, String headerName, String tokenName) {
         String preferValue = null;
-        for (String headerValue : context.getHttpHeaders().getOrDefault(headerName, Collections.emptyList())) {
-            for (String headerValueElement : headerValue.split(",")) {
-                headerValueElement = headerValueElement.split(";")[0];
-                String[] tokenAndValue = headerValueElement.split("=", 2);
-                if (tokenName.equals(tokenAndValue[0].trim())) {
-                    preferValue = "";
-                    if (tokenAndValue.length > 1) {
-                        preferValue = tokenAndValue[1].trim();
+        for (String curHeaderName : context.getHttpHeaders().keySet()) {
+            if (curHeaderName.equalsIgnoreCase(headerName)) {
```

Comment:
might be nice to have a test that exercises the case-insensitive flavor, but looks good to me

Reply:
Yeah, I added integration tests that use various cases (lower/uppercase/etc.), though the case always showed up the same when it reached this code.  That was the best I could come up with.